pkg/models/resource.go

```diff
@@ -195,6 +224,9 @@ func (r *Resources) Merge(other Resources) *Resources {
 	if newR.GPU <= 0 {
 		newR.GPU = other.GPU
 	}
+	if len(newR.GPUs) <= 0 {
```

Comment:
I was about to comment that len(array) is unlikely to ever be less than 0, but then I fell down the rabbit hole of "why does Go return signed values for len()", so I guess this is fine.